File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/creation/Creator.java
package org.javamaster.invocationlab.admin.service.creation;

import org.javamaster.invocationlab.admin.service.GAV;
import org.javamaster.invocationlab.admin.service.Pair;

/**
 * @author yudong
 */
public interface Creator {

    Pair<Boolean, String> create(String cluster, GAV gav, String serviceName);

    Pair<Boolean, String> refresh(String cluster, String serviceName);

    String getLatestVersion(GAV gav);
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/creation/Creator.kt
package org.javamaster.invocationlab.admin.service.creation

import org.javamaster.invocationlab.admin.service.GAV
import org.javamaster.invocationlab.admin.service.Pair

/**
 * @author yudong
 */
interface Creator {
    fun create(cluster: String, gav: GAV, serviceName: String): Pair<Boolean, String>

    fun remove(cluster: String, serviceName: String)

    fun refresh(cluster: String, serviceName: String): Pair<Boolean, String>

    fun getLatestVersion(gav: GAV): String
}
